Sources/tart/Data+Chunks.swift

```diff
@@ -0,0 +1,18 @@
+import Foundation
+
+extension Data {
+  /*
+   * Performant version of splitting a Data into chunks of a given size.
+   * It appers that "Data.chunks` is not as performant as chunking the range of the data
```

Comment:
How did you measure this? Have you used "CPU Profiler" of Instruments? Have you ran `sudo purge` before the test?

I've replaced the `tart list`'s `run()` function with the following contents:

```swift
let disk = try FileHandle(forReadingFrom: URL(fileURLWithPath: "/Users/edi/.tart/vms/macos/disk.img"))

let holeGranularityBytes = 64 * 1024

var count = 0

while true {
  guard let data = try disk.read(upToCount: 64 * 1024 * 1024) else {
    break
  }
  
  for chunk in data.chunks(ofCount: holeGranularityBytes) {
    count += chunk.count
  }
}

print(count)
```

Here's a profile of `chunks()` which takes ~15 seconds and ~20 Mc to churn through macOS Sonoma's disk image:

<img width="1624" alt="Screenshot 2024-08-05 at 15 58 08" src="https://github.com/user-attachments/assets/3e3e28ad-b206-4fd3-93c9-603cf7404eaa">

Now by changing `chunks()` to `subdataChunks()` we do the same with ~27 seconds and using ~110 Mc:

<img width="1624" alt="Screenshot 2024-08-05 at 16 00 17" src="https://github.com/user-attachments/assets/05e62eeb-e32e-4deb-b812-0c566429732d">

Reply:
I didn't purge but this was not the biggest optimization. Let's revert and profile separately. See [773a403](https://github.com/cirruslabs/tart/pull/878/commits/773a40301136482a9f7103048f8441b7bb63b429)